TATB-36- Redesign login and registration page

PrabhatJTDeveloper opened the pull request.

PrabhatJTDeveloper committed:
TATB - 30: When click skip button , pop up will remain even after question change
 -Give camelCase to variables
 -Fixed bug when click skip button the popup also swipes on timer end
 -Removed a console error

PrabhatJTDeveloper committed:
TATB:30 :
  -reverted chagnes
  - removed unwated code

PrabhatJTDeveloper committed:
TATB-32- Fixed conflict in Redesign login and registration page

PrabhatJTDeveloper committed:
TATB - 36 - Fix the issue of save register user data

charmijtdev commented:
Please change name of PR

PrabhatJTDeveloper committed:
TATB - 29 : Registration page should contain header (#42)

* Technical assistant : Implement header for all pages

* TATB - 29 : Move inline CSS to external SCSS file

* TATB - 29 : Remove inline css and Remove popup box model in header section

* TATB - 29 :  Fixed Merge Conflicts

---------

Co-authored-by: Dipesh Gandhi <dipesh_gandhi9@gmail.com>
Co-authored-by: Charmi Chauhan <88572890+charmijtdev@users.noreply.github.com>

PrabhatJTDeveloper committed:
TATB- 33 : login page registration page flow change functionality (#41)

* TATB-05-BUG-Add-login-link-to-user-registration-form-login-form-component

* TATB-05-BUG-Add-login-link-to-user-registartion-form-login-form-component

* TATB - 33 : Login page Registration page flow change functionality

* TATB - 33 : Login page Registration page flow change functionality

* TATB - 33 : Login page Registartion page flow change functionality - changes - Login component,Registration component

* TATB - 33 : Login page Registration page flow change functionality changes - app module, registration component,toast component

* TATB - 33 : Login page Registration page flow change functinality changes - login, registration components

PrabhatJTDeveloper committed:
TATB-  38 : remove auth service use to auth guard for authentication (#43)

* TATB - 38 : Remove Auth service use to Auth guard for authentication

* TATB - 38 : Remove Auth service use to Auth guard for authentication changes-auth guards,login,header and question service name to chnage authentication service.

* TATB - 38 : Remove Auth service use to Auth guard for authentication changes - auth guard,login component,header, authentication service.

* TATB - 38 : Remove Auth service use to Auth guard for authentication.  changes - result component and authentication service

PrabhatJTDeveloper committed:
TATB - 40 : change design for confirmation dialog box (#44)

* TATB - 40 : change design for confirmation dialog box.

* TATB - 40 : change design for confirmation dialog box changes- modal component , login

* TATB - 40 : change design for confirmation dialog box - change login and registration components

charmijtdev commented on src/app/authorization/login/login.component.html:
```diff
@@ -1,92 +1,62 @@
+            id="email"
+            formControlName="email"
+          />
+          <div class="form-text text-danger" *ngIf="">Email is required.</div>
```
why ng if is empty

PrabhatJTDeveloper replied:
ok I have uses to   *ngIf validation email id

charmijtdev commented on src/app/authorization/login/login.component.html:
```diff
@@ -1,92 +1,62 @@
+            formControlName="email"
+          />
+          <div class="form-text text-danger" *ngIf="">Email is required.</div>
+          <div class="form-text text-danger" *ngIf="">Email is Invalid.</div>
```
same here

PrabhatJTDeveloper replied:
Ok I have done *ngIf validation add

charmijtdev commented:
Please add condition for all validation there is no validation at all

PrabhatJTDeveloper committed:
TATB - 37 : Redesign Quiz Test Component (#45)

* - TATB-37 redesign the quiz test template
  - for that change the html and scss file of the quiz component .

* - TATB 37
   - removed unnecessary id and role in the button from the html file
   - removed unwanted css in scss file

* - TATB 37
   - removed unwanted id and class from carousel in html file
   - removed the css which is also used as boostrap class.

* - TATB 37 removed unused div in html file

* - TATB 37
  - make change in condition of hightlighted text in quiz.component.html file

* - TATB 37
   - give the boostrap class and remove from the css
   - and make change in html file

* - TATB 37
   - make changes in html and scss file.

* - TATB 37
   - removed "this" from the {{this.selectedQuizType}} in the quiz component html file.

* - TATB 37
  - removed "this" from the html file

PrabhatJTDeveloper committed:
TATB-36-Add validation in login page

charmijtdev commented on src/app/authorization/login/login.component.html:
```diff
@@ -1,92 +1,85 @@
+              Password is required.
+            </span>
+            <span
+              *ngIf="loginFormValidator['password'].errors?.['pattern'] || loginFormValidator['password'].errors?.['maxlength'] && (loginFormValidator['password'].dirty || loginFormValidator['password'].touched)"
```
please wrap condition inside the () like (loginFormControl['password']?.errors?.['required'] && (loginFormControl['password']?.dirty || loginFormControl['password']?.touched)) in every condition

PrabhatJTDeveloper replied:
I  have Add () every conditions

charmijtdev commented on src/app/authorization/login/login.component.html:
```diff
@@ -1,92 +1,85 @@
+      </div>
+      <div class="text-center">
+        <h6 class="mt-3">
+          Don't Have an Account ?<a href="/registration"> Register Now</a>
```
why href tag ? as before routerlink was used . Please change this

PrabhatJTDeveloper replied:
I have Remove href tag. I have Add router link

charmijtdev commented on src/app/authorization/login/login.component.scss:
```diff
@@ -1,16 +1,41 @@
-  font-size: calc(70% + 1vh);
-  margin-left: 5px;
+.card.shadow {
+  width: 25rem;
```
why width ? manage with col class

PrabhatJTDeveloper replied:
ok I have remove CSS card-shadow. I have Add Bootstrap class in card-shadow

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -4,6 +4,7 @@ import { Router } from '@angular/router';
+
 @Component({
   selector: 'app-login',
   templateUrl: './login.component.html',
```
please remove comma after  styleurls styleUrls: ['./login.component.scss'],

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -46,24 +50,30 @@ export class LoginComponent implements OnInit, OnDestroy {
     this.createForm();
-    if (localStorage.getItem('registeruser')?.length) {
-      let data: any = localStorage.getItem('registeruser');
+    if (localStorage.getItem('registerUser')?.length) {
```
No need to check length please remove this

charmijtdev commented on src/app/authorization/login/login.component.ts:
```diff
@@ -46,24 +50,30 @@ export class LoginComponent implements OnInit, OnDestroy {
   }
 
-  get loginFormControl() {
+  get loginFormValidator() {
```
please change this to loginFormControllers

charmijtdev commented on src/app/authorization/registration/registration.component.html:
```diff
@@ -1,237 +1,232 @@
-        <div class="row">
-          <div class="col text-center p-2 mb-3">
-            <span class="fs-3">User Registration</span>
+<div class="card shadow col-lg-4 col-md-4 col-sm-8 col-xs-8 mt-5">
```
same here wrap conditions to ()

PrabhatJTDeveloper replied:
I have Remove conditions only  for using col-sm-3 and mt-5

charmijtdev commented on src/app/authorization/registration/registration.component.html:
```diff
@@ -1,237 +1,232 @@
-                formControlName="confirmPassword"
+                type="radio"
+                class="form-check-input"
+                value="f"
```
please check value before this change

PrabhatJTDeveloper replied:
I have change value

charmijtdev commented on src/app/authorization/registration/registration.component.html:
```diff
@@ -1,237 +1,232 @@
+            type="date"
+            class="form-control"
+            formControlName="dateOfBirth"
+            [max]="todayDate"
```
please add keydown function here

PrabhatJTDeveloper replied:
ok I have Add keydown function

charmijtdev commented on src/app/authorization/registration/registration.component.html:
```diff
@@ -1,237 +1,232 @@
+        <div class="col-mb-3">
+          <div class="d-grid">
+            <button
+              type="button"
```
please change this type to submit and do function calling on ngSubmit

PrabhatJTDeveloper replied:
Ok I have change type of button to submit button

charmijtdev commented on src/app/authorization/registration/registration.component.html:
```diff
@@ -1,237 +1,232 @@
+
+      <div class="text-center">
+        <h6 class="mt-3">
+          Already have an account?<a href="/login"> Login</a>
```
why herf tag ? please change this

PrabhatJTDeveloper replied:
I have remove href tag using router link

PrabhatJTDeveloper committed:
TATB - 39 [BUG]: Wrap Dialog Data into one config object for more readability (#46)

* - TATB 39
    - for wraping dialog data into one config object changed the modal component ts file.
    - in modal component html file use that object .
    - and the modal is opening in signout which is in header component and skip button which is in quiz component so changed this two components.
    - add one object signoutModal in dialogData.json file.

* - TATB 39
     - removed ?.length from the login component ts file.
     - removed import of dilogData.json from dialog service
     - updated modal component html file.

* - TATB 39
    - removed inline css and give the extenal css
    - added the userId in the login component ts file
    - removed unnecassary line space and comma from the dialog service ts file
    - formated the code in modal component html file and also removed commented code.

* - TATB 39
    - Removed extra comma from the login and header component ts file and also change the function name for opening dialog modal in header component ts file.
    - change the class name in header component scss file.
    - make change in dialogData.json file

PrabhatJTDeveloper committed:
TATB-36 : Change href tag to router link  chnage value and Add keydown function in login and registration page

PrabhatJTDeveloper committed:
TATB-36 : Fix conflict in Redesign login and registration page

PrabhatJTDeveloper committed:
TATB-36 : Responsive login and Registration page
